configuration/components/loki.libsonnet

```diff
@@ -492,6 +523,45 @@ function(params) {
     str
   ),
 
+  // buildEnvVars compiles a list of parameters whose values are only
+  // known at rollout time. Loki config supports env variable expansion, we
+  // leverage this feature to inject these config values at rollout time.
+  // Some values are only specific to some component but since all components
+  // consume the same ConfigMap we decided for all components share the same
+  // of env variables.
+  local buildConfigVars(component) =
+    // Syntactic sugar
+    local envVarFromValue(name, value) = { name: name, value: value };
+    local envVarFromSecret(name, secretName, secretKey) = { name: name, valueFrom: { secretKeyRef: { name: secretName, key: secretKey } } };
+    local osc = loki.config.objectStorageConfig;
+    local rsc = loki.config.rulesStorageConfig;
+    [
+      envVarFromValue('LOKI_LOG_LEVEL', loki.config.logLevel),
+      // The following values are added with '', because env only supports strings
+      envVarFromValue('LOKI_REPLICATION_FACTOR', '' + loki.config.replicationFactor),
+      envVarFromValue('LOKI_QUERIER_MAX_CONCURRENCY', '' + loki.config.query.concurrency),
+      envVarFromValue('LOKI_INGESTER_WAL_REPLAY_MEMORY_CEILING', '' + loki.config.wal.replayMemoryCeiling),
```

Comment:
Still not sure if this should be considered an int or a string since as far as I saw this can may or may not have the unit suffix (KB, MB, GB) applied [1]. Currently, it's coded as if it was an int. I'll test tomorrow morning to be sure

[1] https://grafana.com/docs/loki/latest/configuration/#ingester

Reply:
It is not a value with unit: https://github.com/rhobs/configuration/blob/main/services/observatorium-logs-template.jsonnet#L80